lib/widgets/message_list.dart

```diff
@@ -235,6 +238,18 @@ class _MessageListPageState extends State<MessageListPage> implements MessageLis
   void initState() {
     super.initState();
     narrow = widget.initNarrow;
+
+    switch (narrow) {
+      case CombinedFeedNarrow():
+      case ChannelNarrow():
+      case TopicNarrow():
+      case DmNarrow():
+        break;
+      case MentionsNarrow():
+      case StarredMessagesNarrow():
+        // TODO(#1583) write tests for this
+        _markReadOnScroll = false;
```

Comment:
Huh, is this really what web does? Even if you have the "Automatically mark messages as read" setting set to "Always"? (If that's either of the other values, then it's as if this were false anyway.)

As I said in the office earlier (probably a bit after you sent this PR), it seems fine to me if choosing "Always" for that setting means always. In particular that was the behavior Zulip web had, with no option to do something else, for a number of years.

In any event I think this isn't a blocker for search. With the default value of the auto-mark-read setting, at "conversations only", it'll have no effect.